apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/Trace/SimpleManagedTracer.cs

```diff
@@ -246,23 +246,24 @@ private void AnnotateSpan(TraceSpan span, Dictionary<string, string> labels)
         public void SetStackTrace(StackTrace stackTrace)
         {
             GaxPreconditions.CheckNotNull(stackTrace, nameof(stackTrace));
-            var currentStack = TraceStack;
-            CheckStackNotEmpty(currentStack);
-
-            AnnotateSpan(currentStack.Peek().TraceSpan, TraceLabels.FromStackTrace(stackTrace));
+            AnnotateSpan(TraceLabels.FromStackTrace(stackTrace));
         }
 
         /// <inheritdoc />
-        public string GetCurrentTraceId() =>_trace.TraceId;
+        public string GetCurrentTraceId() => _trace.TraceId;
         
         /// <inheritdoc />
         public ulong? GetCurrentSpanId() => GetCurrentSpanId(TraceStack);
 
         /// <summary>
         /// Gets the current span id of the specified stack or null if none exists.
         /// </summary>
-        private ulong? GetCurrentSpanId(ImmutableStack<Span> traceStack) =>
-            traceStack.IsEmpty? _rootSpanParentId : traceStack.Peek().TraceSpan.SpanId;
+        private ulong? GetCurrentSpanId(ImmutableStack<Span> traceStack)
+        {
+            traceStack = GetUpdatedStack(traceStack);
```

Comment:
I think I'd actually remove the call to `GetUpdatedStack` here (sorry for the change) and make it the caller's responsibility to get the updated stack and reassign it to `TraceStack`